crates/ruff/src/commands/format.rs

```diff
@@ -250,8 +265,12 @@ pub(crate) fn format_path(
         }
     };
 
+    // Don't write back to the cache if formatting a range.
+    let write_cache = cache.filter(|_| range.is_none());
```

Comment:
I might consider `let cache = cache.filter(|_| range.is_none())` to make sure I didn't miss any usages of `cache` below.

Reply:
Excellent point